components/payment_form/address_form.tsx

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useState} from 'react';
+
+import {FormattedMessage, MessageDescriptor, useIntl} from 'react-intl';
+
+import {Address} from '@mattermost/types/cloud';
+import DropdownInput from 'components/dropdown_input';
+import {COUNTRIES} from 'utils/countries';
+import Input from 'components/widgets/inputs/input/input';
+
+import StateSelector from './state_selector';
+
+import './payment_form.scss';
+
+type AddressFormProps = {
+    onAddressChange: (address: Address) => void;
+    onBlur: () => void;
+    title: MessageDescriptor;
+    formId: string;
+    initialAddress: Address;
+}
+
+const AddressForm = (props: AddressFormProps) => {
+    const {formatMessage} = useIntl();
+    const [address, setAddress] = useState<Address>(props.initialAddress);
+    const handleCountryChange = (option: any) => {
+        setAddress({...address, country: option.value});
+        props.onAddressChange({...address, country: option.value});
+    };
+
+    const handleStateChange = (option: any) => {
+        setAddress({...address, state: option});
+        props.onAddressChange({...address, state: option});
+    };
+
+    const handleInputChange = (
+        event:
+        | React.ChangeEvent<HTMLInputElement>
+        | React.ChangeEvent<HTMLSelectElement>,
+    ) => {
+        const target = event.target;
+        const name = target.name;
+        const value = target.value;
+
+        const newStateValue = {
+            [name]: value,
+        } as unknown as Pick<Address, keyof Address>;
+
+        setAddress({...address, ...newStateValue});
+
+        const {onAddressChange} = props;
+        if (onAddressChange) {
```

Comment:
Nit: We can skip the onAddressChange null check since this is a required prop, and this component can't do anything meaningful without it.